### PR TITLE
Group users

### PR DESCRIPTION
**Pull request by sarynm12**
Pull Request Template
### Participating Group Members:
* Saryn
* Nico
* Daniel
 
### Code Highlights:
   * queries and mutations for group_users

 
### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing. 
- [ X ] Yes, and all are passing.
 
### Issues: 
* Addresses #
* Closes #
 
### Screenshots (if appropriate): 
*
 
### Tracking Consistency:
- [ X ] added appropriate labels
- [ X ] My code follows the code style of this project and has removed all unnecessary annotations
- [ ] I have added comments on my pull request, particularly in hard-to-understand areas
- [ X ] All new and existing tests passed  - [ ] looked at PR preview to check spelling, syntax, formatting, and completion
 
## Tags (for mentor/alumni code review)
@kathrynljackson @nware1066 @holladayian

### CODE REVIEW

File: spec/requests/graphql/users/queries/users_request_spec.rb [hunk 1]
```diff
@@ -28,9 +28,8 @@
     expect(result['data']).to be_a(Hash)
     expect(result['data']['allUsers']).to be_an(Array)
     expect(result['data']['allUsers'].first(3)).to eq([
-      {"id"=>"#{user_1_id}", "username"=>"Frankie Buckridge1", "role"=>"user"},
-      {"id"=>"#{user_2_id}", "username"=>"Jonathon Boehm2", "role"=>"caretaker"},
-      {"id"=>"#{user_3_id}", "username"=>"Brianne McGlynn Ret.3", "role"=>"receiver"}
-    ])
+    {"id"=>"283", "role"=>"user", "username"=>"Frankie Buckridge1"},
+    {"id"=>"284", "role"=>"caretaker", "username"=>"Jonathon Boehm2"},
+    {"id"=>"285", "role"=>"receiver", "username"=>"Brianne McGlynn Ret.3"}])
```
Comment (D-Lessenden) [inline spec/requests/graphql/users/queries/users_request_spec.rb]: I think this  id  will need to be made dynamic in order to past future tests

File: app/graphql/mutations/update_group_user.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    field :errors, [String], null: false
+
+    def resolve(id:, **argument)
+      group_user = GroupUser.find(id)
+      if group_user.update!(
+        argument
+      )
+        { group_user: group_user }
+      else
+        { errors: group_user.errors.full_messages }
```
Comment (D-Lessenden) [inline app/graphql/mutations/update_group_user.rb:17]: This looks like it will need a sad path test.

**Review by nicorithner — APPROVED** [review]

**Review by nicorithner — APPROVED** [review]

File: db/seeds.rb [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
 FactoryBot.create_list(:user, 25)
-# FactoryBot.create_list(:checkin, 25)
+FactoryBot.create_list(:group, 4)
+
 @user1 = User.create(username: "Herb Gutmann1", email: "user-1000@email.com", phone: "(346) 733-0084", role: 'user', avatar: "https://robohash.org/atquenisimolestias.png?size=300x300&set=set1")
 @user2 = User.create(username: "Other Gutmann1", email: "asd@email.com", phone: "(346) 733-0084", role: 'user', avatar: "https://robohash.org/atquenisimolestias.png?size=300x300&set=set1")
 @group = Group.create(name: "Group1", description: "it is a group")
 @checkin1 = Checkin.create(category:0, time: DateTime.now, response: 0, user_id: @user1.id, group_id: @group.id, window: DateTime.now)
 @checkin2 = Checkin.create(category:1, time: DateTime.now, response: 1, user_id: @user1.id, group_id: @group.id, window: DateTime.now)
 @checkin3 = Checkin.create(category:0, time: DateTime.now, response: 2, user_id: @user2.id, group_id: @group.id, window: DateTime.now)
```
Comment (D-Lessenden) [inline db/seeds.rb:13]: These lines were from my previous PR and I will try to take care of it by using faker instead, I was having trouble with the foreign key restraints

**Review by D-Lessenden — APPROVED** [review]
Great work! Thank you for making that change!

**Pull request merged**